cmd/srcd/cmd/gitbase_back_comp_test.go

```diff
@@ -0,0 +1,222 @@
+// +build integration
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	cmdtest "github.com/src-d/engine/cmd/test-utils"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/stretchr/testify/suite"
+)
+
+type GitbaseBackCompTestSuite struct {
+	cmdtest.IntegrationBackCompSuite
+	timeout     time.Duration
+	testDir     string
+	testDataDir string
+}
+
+func TestGitbaseBackCompTestSuite(t *testing.T) {
+	s := GitbaseBackCompTestSuite{
+		IntegrationBackCompSuite: cmdtest.NewIntegrationBackCompSuite(),
+		timeout:                  1 * time.Minute,
+	}
+	suite.Run(t, &s)
+}
+
+func (s *GitbaseBackCompTestSuite) createRepo(name string) {
+	s.T().Helper()
+	require := s.Require()
+
+	repoPath := filepath.Join(s.testDir, name)
+	err := os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+}
+
+func (s *GitbaseBackCompTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "gitbase-back-comp-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	homedir, err := homedir.Dir()
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "unable to get home dir"))
+	}
+
+	s.testDataDir = filepath.ToSlash(filepath.Join(homedir, ".srcd", "gitbase"))
+	os.RemoveAll(s.testDataDir)
+
+	for i := range [3]int{} {
+		s.createRepo(fmt.Sprintf("repo-%d", i))
+	}
+}
+
+func (s *GitbaseBackCompTestSuite) TearDownTest() {
+	s.Cmd.RunStop(context.Background())
+	s.PrevCmd.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+	os.RemoveAll(s.testDataDir)
+}
+
+var newLineFormatter = regexp.MustCompile(`(\r\n|\r|\n)`)
+var reposExtractors = regexp.MustCompile(`(?:\|\s+([\w-]+)\s+\|\n)`)
+
+func (s *GitbaseBackCompTestSuite) getBufString(buf *bytes.Buffer) string {
```

Comment:
this is the same as `normalizeNewLine` in `common.go`?

Reply:
addressed